presto-main/src/test/java/com/facebook/presto/operator/GroupByHashYieldAssertion.java

```diff
@@ -120,14 +120,17 @@ public static GroupByHashYieldResult finishOperatorWithYieldingGroupByHash(List<
 
             long newMemoryUsage = operator.getOperatorContext().getDriverContext().getMemoryUsage();
 
-            // We are below the 1MB GUARANTEED_MEMORY limit (also need some buffer for aggregator).
-            // For such cases, the operator is blocked by memory
-            if (newMemoryUsage < (1 << 20) + additionalMemoryInBytes) {
-                // assert non-blocking
-                assertTrue(operator.needsInput());
-                assertTrue(operator.getOperatorContext().isWaitingForMemory().isDone());
+            // Skip if the memory usage is not large enough since we cannot distinguish
+            // between rehash and memory used by aggregator
+            if (newMemoryUsage < additionalMemoryInBytes * 3) {
```

Comment:
This looks quite bit coarse. For example, `HashSemiJoinOperator` does not have an aggregator. The memory overhead won't be this huge.